vpr/src/route/router_lookahead.h

```diff
@@ -8,6 +8,9 @@ struct t_conn_cost_params; //Forward declaration
 class RouterLookahead {
   public:
     virtual float get_expected_cost(int node, int target_node, const t_conn_cost_params& params, float R_upstream) const = 0;
+    virtual float get_expected_delay(int node, int target_node, const t_conn_cost_params& params, float R_upstream) const = 0;
```

Comment:
Because of how the map lookahead works (see https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/master/vpr/src/route/router_lookahead_map.cpp#L238), this method should be `get_expected_delay_and_cong` and allow for the RouterLookahead object to provide both values at the same time.  If you look at how `get_expected_cong` is used, it is always paired with `get_expected_delay`, and lookaheads like the map lookahead would have to lookup the matrix index twice, and perform the memory access twice, which would be terribly expensive.

@kmurray

Reply:
Given that the classic lookahead also performs the same memory access for `get_expected_delay` and `get_expected_cong`, there would also be memory access saving in that case.